File: stack/RestrictiveCandyCrush.cpp
class Solution{
    public:
    string Reduced_String(int k,string s){
        // Your code goes here
        if(k == 1){
            return "";
        }
        string output = "";
        stack<pair<char,int>> stk;
        for(int i=0;i<s.length();i++){
            if(stk.size() == 0){
                stk.push(make_pair(s[i],1));
            }
            else if(stk.top().first != s[i]){
                stk.push(make_pair(s[i],1));
            }
            else{
                if(stk.top().second == k-1){
                    stk.pop();
                }
                else{
                    stk.top().second+=1;
                }
                
            }
        }
        while(!stk.empty()){
            while(stk.top().second>0){
                output+=stk.top().first;
                stk.top().second-=1;
            }
            stk.pop();
        }
        reverse(output.begin(),output.end());
        return output;
    }


};

File: stack/RestrictiveCandyCrush.py
class Solution:
    def Reduced_String(self, k, s):
        # Your code goes here
        # return the reduced string
        stack = []
        if k == 1:
            return ''
        for i in range(len(s)):
            if len(stack) == 0:
                stack.append([s[i],1])
            elif stack[-1][0]!=s[i]:
                stack.append([s[i],1])
            else:
                if stack[-1][1]==k-1:
                    stack.pop(-1)
                else:
                    stack[-1][1]+=1
        return ''.join(x[0]*x[1] for x in stack)
